### PR TITLE
Make standalone to advertise "localhost" to avoid failure when hostname is not resolvable

### PR DESCRIPTION
**Pull request by merlimat**
### Motivation

The standalone service is failing to startup when the hostname is not resolvable to an IP. This is common in laptops. 

The reason is that the function worker in standalone is trying to connect to the advertised address rather than "localhost". 

To fix this, we should make the standalone to always advertise "localhost" as the the broker address. This is ok because in any case the standalone treats the advertisement as the pulsar proxy, in that it forces the clients to always go through the same service url, so a client connecting from a different host won't have problems.

### CODE REVIEW

File: pulsar-broker/src/main/java/org/apache/pulsar/PulsarStandaloneStarter.java [hunk 1]
```diff
@@ -67,7 +67,7 @@ public PulsarStandaloneStarter(String[] args) throws Exception {
             zkServers = this.getAdvertisedAddress();
         } else if (isBlank(config.getAdvertisedAddress())) {
             // Use advertised address as local hostname
-            config.setAdvertisedAddress(ServiceConfigurationUtils.unsafeLocalhostResolve());
+            config.setAdvertisedAddress("localhost");
```
Comment (sijie) [inline pulsar-broker/src/main/java/org/apache/pulsar/PulsarStandaloneStarter.java]: I feel it is better to try to add a method to `ServiceConfigurationUtils` to resolve localhost first. It is not resolved then fallback to use "localhost".

**Comment by codelipenghui** [comment]
run java8 tests

**Comment by aahmed-se** [comment]
run java8 tests

**Comment by sijie** [comment]
run java8 tests

**Comment by sijie** [comment]
run java8 tests

**Comment by sijie** [comment]
run java8 tests

**Comment by sijie** [comment]
run java8 tests

**Comment by sijie** [comment]
run java8 tests

**Comment by sijie** [comment]
run java8 tests

**Comment by sijie** [comment]
run java8 tests

**Comment by sijie** [comment]
run java8 tests

**Comment by sijie** [comment]
retest this please

**Comment by sijie** [comment]
run java8 tests

**Comment by tuteng** [comment]
run java8 tests

**Comment by tuteng** [comment]
run java8 tests

**Comment by wolfstudy** [comment]
run java8 tests

**Pull request merged**